rusk-wallet/src/lib/prompt.rs

```diff
@@ -371,26 +371,14 @@ fn is_valid_addr(addr: &str) -> bool {
 }
 
 fn check_valid_denom(num: f64, balance: f64) -> Result<(), String> {
-    if num.is_finite() && num > 0.0 {
-        if num > MAX_CONVERTIBLE {
-            return Err(format!(
-                "This is greater than the max amount allowed {}",
-                MAX_CONVERTIBLE
-            ));
-        }
-        if num < MIN_CONVERTIBLE {
-            return Err(format!(
-                "Input too low, please increase at least to {}",
-                MIN_CONVERTIBLE
-            ));
-        }
-        if num < balance {
-            Ok(())
-        } else {
-            Err("insufficient balance".to_owned())
+    let min = MIN_CONVERTIBLE;
+    let max = f64::min(balance, MAX_CONVERTIBLE);
```

Comment:
It should be an interesting situation when the balance is larger than the maximum allowable. I don't think it will ever happen, but it's a nice defense.